Modules/modExtendSaveAsText.ACM

```diff
@@ -65,15 +66,49 @@ Public Sub SaveAsText(objType As asObjectType, objName As String, strFilename As
   Ignores(7).bytsStart = "NameMap = Begin" & vbCrLf
   Ignores(7).bytsEnd = "End" & vbCrLf
   
+  ' work out whether we need to convert the file to UTF8
+  ' initially set to the user selection
+  blnSaveAsUTF8 = Forms("Access SVN - Options").chkOutputUTF8
+  ' For Forms, Report, Macros and Queries the output from the built in SaveAsText is in the 2 byte UCS2 format
+  ' on later versions of MS Access, otherwise it is in ANSI
+  ' If we already have this file in the repository and it is in UTF-8 we should save it as UTF-8 anyway
+  Select Case objType
+    Case asObjectType.acForm, asObjectType.acReport, asObjectType.acQuery, asObjectType.acMacro
+      If FileExists(strFilename) Then
+        If IsUTF8(strFilename) Then
+          ' Fileis already in the repository and it's in UTF8 so save in that format now overriding user selection
+          blnSaveAsUTF8 = True
+        End If
+      End If
+  End Select
+      
   Select Case objType
     Case asObjectType.acForm, asObjectType.acReport
       Application.SaveAsText objType, objName, strFilename & ".tmp"
-      If Not CleanFormReportFile(strFilename & ".tmp", strFilename, Ignores) Then
-        MsgBox "Something went wrong cleaning the file for " & objName & "."
+      If IsUCS2(strFilename & ".tmp") Then
+        If CleanFormReportFile(strFilename & ".tmp", strFilename, Ignores) Then
+          If blnSaveAsUTF8 And IsUCS2(strFilename) Then
+            If Not ConvFileUCS2_to_UTF8(strFilename) Then 'convert file to UTF8
+              MsgBox "Unable to convert '" & objName & "' to UTF-8 encoding. Beware when committing."
+            End If
+          End If
+        Else
+          MsgBox "Something went wrong cleaning the file for " & objName & "."
+        End If
+      Else
+        ' This version of Access doesn't produce UCS2 output
+        Name strFilename & ".tmp" As strFilename
       End If
       If FileExists(strFilename & ".tmp") Then Kill strFilename & ".tmp"
-    Case asObjectType.acQuery, asObjectType.acMacro, asObjectType.acModule
+    Case asObjectType.acQuery, asObjectType.acMacro
       Application.SaveAsText objType, objName, strFilename
+      If blnSaveAsUTF8 And IsUCS2(strFilename) Then

```

Comment:
Should this not be `blnSaveAsUTF8 And Not IsUTF8(strFilename)` then warn the user if the file is not UCS2/cannot be converted to UTF8?